### PR TITLE
Document HADOOP_SNAPPY as a supported compression format for aws_kinesis_firehose_delivery_stream

### PR DESCRIPTION
**Pull request by bnix**
The [Firehose SDK supports Hadoop Snappy](https://github.com/aws/aws-sdk-go/blob/main/service/firehose/api.go#L8216-L8242) as a compression format.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: website/docs/r/kinesis_firehose_delivery_stream.html.markdown [hunk 1]
```diff
@@ -410,7 +410,7 @@ The `s3_configuration` object supports the following:
 * `buffer_size` - (Optional) Buffer incoming data to the specified size, in MBs, before delivering it to the destination. The default value is 5.
                                 We recommend setting SizeInMBs to a value greater than the amount of data you typically ingest into the delivery stream in 10 seconds. For example, if you typically ingest data at 1 MB/sec set SizeInMBs to be 10 MB or higher.
 * `buffer_interval` - (Optional) Buffer incoming data for the specified period of time, in seconds, before delivering it to the destination. The default value is 300.
-* `compression_format` - (Optional) The compression format. If no value is specified, the default is UNCOMPRESSED. Other supported values are GZIP, ZIP & Snappy. If the destination is redshift you cannot use ZIP or Snappy.
+* `compression_format` - (Optional) The compression format. If no value is specified, the default is `UNCOMPRESSED`. Other supported values are `GZIP`, `ZIP`, `Snappy`, & `HADOOP_SNAPPY`. If the destination is redshift you cannot use `ZIP`, `Snappy`, or `HADOOP_SNAPPY`.
```
Comment (anGie44) [inline website/docs/r/kinesis_firehose_delivery_stream.html.markdown:413]: Just wanted to double-check: AWS api docs say only `Snappy` and `ZIP` are not supported when using redshift (seen here: https://docs.aws.amazon.com/firehose/latest/APIReference/API_S3DestinationConfiguration.html#Firehose-Type-S3DestinationConfiguration-CompressionFormat) , but that could be incorrect on their part..have you found that `HADOOP_SNAPPY` is also not supported?

**Review by anGie44 — APPROVED** [review]
Overall looks great, thank you for this update @bnix ! Just a small question to verify the not-supported types

File: website/docs/r/kinesis_firehose_delivery_stream.html.markdown (hunk 1 above)
Comment (bnix) [reply website/docs/r/kinesis_firehose_delivery_stream.html.markdown:413]: Thanks for asking. I should maybe remove this given the doc you linked differs, however when I used the AWS console to create a new stream targeting my Redshift cluster the only compression option I could pick was GZIP.

Comment (anGie44) [reply website/docs/r/kinesis_firehose_delivery_stream.html.markdown:413]: HI @bnix, apologies for the delay! On my end, I've tested using all the supported `compression_format` values with a `redshift` destination, and strangely the configurations apply successfully without error as suggested by the AWS API docs (it's possible the SDK supports values not visible in the console 🙃 ). I'll update this accordingly (if the git branch is writeable :) )

Comment (anGie44) [inline website/docs/r/kinesis_firehose_delivery_stream.html.markdown:413]:
```suggestion
* `compression_format` - (Optional) The compression format. If no value is specified, the default is `UNCOMPRESSED`. Other supported values are `GZIP`, `ZIP`, `Snappy`, & `HADOOP_SNAPPY`.
```

**Pull request merged**